tests/integration_tests/vectorstores/test_weaviate.py

```diff
@@ -81,6 +81,23 @@ def test_similarity_search_with_metadata_and_filter(
         )
         assert output == [Document(page_content="foo", metadata={"page": 0})]
 
+    @pytest.mark.vcr(ignore_localhost=True)
+    def test_similarity_search_with_metadata_and_additional(
+        self, weaviate_url: str, embedding_openai: OpenAIEmbeddings
+    ) -> None:
+        """Test end to end construction and search with metadata and additional."""
+        texts = ["foo", "bar", "baz"]
+        metadatas = [{"page": i} for i in range(len(texts))]
+        docsearch = Weaviate.from_texts(
+            texts, embedding_openai, metadatas=metadatas, weaviate_url=weaviate_url
+        )
+        output = docsearch.similarity_search(
+            "foo",
+            k=2,
+            additional=["certainty"],
```

Comment:
is certainty supposed to be in the output? maybe im misunderstanding what with_additional does

Reply:
You are absolutely right. I am trying to run the integration tests myself. But I cannot make it work. Now that you mention it, I think the output check should be like this:

`assert output == [Document(page_content="foo", metadata={"page": 0, "_additional": {"certainty": 0}})]`

I do not know how to get the value for certainty in the test.

Reply:
as in you can't make the integration test pass or you can't make it run at all? to run you can do
```bash
pytest tests/integration_tests/vectorstores/test_weaviate.py
```

you'll want to spin up local weaviate container, instructions for that at top of integration test file 
```bash
cd tests/integration_tests/vectorstores/docker-compose
docker compose -f weaviate.yml up
```

Reply:
Thanks, learning about poetry at the same time. I managed to get the integration tests running with your tips. Now it keeps telling me he is missing the OpenAI key:

E           ValueError: Error during query: [{'locations': [{'column': 6, 'line': 1}], 'message': 'explorer: get class: vectorize params: vectorize params: vectorize params: vectorize keywords: remote client vectorize: OpenAI API Key: no api key found neither in request header: X-OpenAI-Api-Key nor in environment variable under OPENAI_APIKEY', 'path': ['Get', 'LangChain_47555ec10c224f01abb7b82f7458c0b6']}]

Trying to figure out what to do here. I added the .env file in the integration_tests folder

Reply:
the integration test requires OpenAI model. if you have an openai api key just set that in the terminal you're running this in with 
```bash
export OPENAI_API_KEY="<api_key>"
```
if you don't have openai key i can just run the test

Reply:
Sorry, I really tried everything I could,  but I cannot fix it. If you can try the test, then I will try to fix it later for my environment.